core/src/main/scala/org/broadinstitute/dsde/rawls/monitor/workspace/runners/LandingZoneCreationStatusRunner.scala

```diff
@@ -0,0 +1,127 @@
+      case _ => "No landing zone, job report, or errors returned from landing zone creation"
+    }
+    Some(s"Landing Zone creation failed: $msg")
+  }
+
+  override def run(
+    job: WorkspaceManagerResourceMonitorRecord
+  )(implicit executionContext: ExecutionContext): Future[JobStatus] = {
+    val billingProjectName = job.billingProjectId match {
+      case Some(name) => RawlsBillingProjectName(name)
+      case None =>
+        logger.error(
+          s"Job to monitor AzureLandingZoneResult created with id ${job.jobControlId} but no billing project set"
+        )
+        return Future.successful(Complete) // nothing more this runner can do with it
+    }
+    val userEmail = job.userEmail match {
+      case Some(email) => email
+      case None =>
+        logger.error(
+          s"Job to monitor AzureLandingZoneResult for billing project $billingProjectName created with id ${job.jobControlId} but no user email set"
+        )
+        val errorMsg =
+          s"Unable to update ${billingProjectName.value} with landing zone status because no user email is stored on monitoring job"
+        billingRepository.updateCreationStatus(billingProjectName, CreationStatuses.Error, Some(errorMsg))
+        return Future.successful(Complete)
+    }
+    getUserCtx(userEmail)
+      .map(userCtx =>
+        Try(workspaceManagerDAO.getCreateAzureLandingZoneResult(job.jobControlId.toString, userCtx)) match {
+          case Failure(exception) =>
+            val message = Some(s"Api call to get landing zone from workspace manager failed: ${exception.getMessage}")
+            billingRepository.updateCreationStatus(billingProjectName, CreationStatuses.Error, message)
+            Incomplete
+          case Success(result) if result.getJobReport != null =>
+            result.getJobReport.getStatus match {
+              // the job just isn't done yet - return indicates this runner isn't finished with the job so it's rescheduled
+              case JobReport.StatusEnum.RUNNING => Incomplete
+              // a normal success
+              case JobReport.StatusEnum.SUCCEEDED if result.getLandingZone != null =>
+                billingRepository.updateCreationStatus(billingProjectName, CreationStatuses.Ready, None)
+                billingRepository.updateLandingZoneId(billingProjectName, result.getLandingZone.getId)
+                Complete
+              // set the error, and indicate this runner is finished with the job
+              case _ =>
+                billingRepository
+                  .updateCreationStatus(billingProjectName, CreationStatuses.Error, failureMessage(result))
+                Complete
+            }
+          // this case is only hit if the job report is null but the landing zone is present in the response
+          case Success(result) if result.getLandingZone != null && result.getLandingZone.getId != null =>
```

Comment:
I'm confused about this case. How would you know it's complete? And I don't think we need to worry about updating the ID.

(FWIW, the ID may always be present now)

Reply:
I'm not sure we should consider this a success case-- how do we know the creation is complete?